src/memory_manager.rs

```diff
@@ -108,8 +108,6 @@ pub fn alloc<VM: VMBinding>(
     // their object sizes are all larger than MMTk's min object size, so we simply put an assertion here.
     // If you plan to use MMTk with a VM with its object size smaller than MMTk's min object size, you should
     // meet the min object size in the fastpath.
-    #[cfg(debug_assertions)]
-    crate::util::forwarding_word::check_alloc_size::<VM>(size);
```

Comment:
We now ask the binding to give us the spec for forwarding bits and forwarding pointer.
I thought we don't need this check in mmtk-core anymore

Reply:
We still make assumptions about the min object size (especially in side metadata specs). This assertion would still be useful. I don't think we have any other code to check the min obj size assumption.

Reply:
We always assume the minimum object size in side metadata specs is 1 word.
There is no other place with such an assumption that I'm aware of.

For per-object side metadata, the bindings can specify a min object size of one word or larger.
For other side metadata, e.g. alloc-bit, the side metadata may still save memory space if it could somehow become aware of the min object sizes of larger than one word, but it still will work if the min object size is one word.

So, we can only check against the constant min object size of one word internally.

Reply:
Search for `MIN_OBJECT_SIZE`. If we no longer use it, we can remove it and we no longer make any assumption about min object size.

Reply:
We don't use `MIN_OBJECT_SIZE` anywhere, but we use `LOG_MIN_OBJECT_SIZE` in several places.
The `check_alloc_size()` function which I removed was VM-dependent, but `LOG_MIN_OBJECT_SIZE` is not.
Still, I will replace the function with a debug-check against `LOG_MIN_OBJECT_SIZE`.